### PR TITLE
PUB-138: Allow proxy rate limit to be controlled remotely

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: proxy/src/main/java/com/wavefront/agent/PushAgent.java [hunk 1]
```diff
@@ -691,6 +690,23 @@ protected void processConfiguration(AgentConfiguration config) {
         logger.fine("Proxy push batch set to (locally) " + pushFlushMaxPoints.get());
       }
 
+      if (config.getCollectorSetsRateLimit() != null && config.getCollectorSetsRateLimit()) {
```
Comment (sushantdewan123) [inline proxy/src/main/java/com/wavefront/agent/PushAgent.java]: Can use - org.apache.commons.lang.BooleanUtils.isTrue(config.getCollectorSetsRateLimit())

File: proxy/src/main/java/com/wavefront/agent/PushAgent.java [hunk 2]
```diff
@@ -691,6 +690,23 @@ protected void processConfiguration(AgentConfiguration config) {
...
+        Long collectorRateLimit = config.getCollectorRateLimit();
+        if (pushRateLimiter != null && collectorRateLimit != null && pushRateLimiter.getRate() != collectorRateLimit) {
+          pushRateLimiter.setRate(collectorRateLimit);
+          logger.info("Proxy rate limit set to " + collectorRateLimit + " remotely");
+        }
+      } else {
+        if (pushRateLimiter != null && pushRateLimiter.getRate() != pushRateLimit) {
+          pushRateLimiter.setRate(pushRateLimit);
+          if (pushRateLimit >= 10_000_000) {
+            logger.info("Proxy rate limit no longer enforced");
```
Comment (sushantdewan123) [inline proxy/src/main/java/com/wavefront/agent/PushAgent.java]: warning? (feel free to ignore my suggestion)

File: proxy/src/main/java/com/wavefront/agent/PushAgent.java (hunk 1 above)
Comment (basilisk487) [reply proxy/src/main/java/com/wavefront/agent/PushAgent.java]: done

File: proxy/src/main/java/com/wavefront/agent/PushAgent.java (hunk 2 above)
Comment (basilisk487) [reply proxy/src/main/java/com/wavefront/agent/PushAgent.java]: agreed - rate limit changes are more important than regular info messaged. done.

**Pull request merged**